### DISCUSSION TITLE
Accessing COGs from Azure Blob Storage

### DISCUSSION THREAD

**Discussion by charalamm** [developmentseed/eoAPI#124] [Q&A]
#### Problem description

I have an azure blob storage account with the COG files and an eoapi raster runtime deployment. When anonymous read access is allowed on the blobs then everything works fine but when there is a requirement for credentials then the API returns 500.

To make the raster eoAPI work with the private blob storage I added the `AZURE_STORAGE_CONNECTION_STRING` environment variable but doesn't seem to solve the problem. At the same time I tried the following snippet on "production" and rasterio works with that environment variable.
```
import os
import rasterio

os.environ["AZURE_STORAGE_CONNECTION_STRING"] = "<>"

blob = "az://data/raster_file.tif"

with rasterio.open(blob, "r") as f:
    print(f.crs)
```

#### Environment Information
The raster runtime is deployed using the raster-uvicorn docker image as an azure container app.
The rasterio version in the container is 1.3.8

**Comment by vincentsarago** [comment]
@charalamm I'm not 💯 sure to fully understand so I'll make some guesses 

you're saying that when you use `az://` url with `AZURE_STORAGE_CONNECTION_STRING` env, rasterio works 👍 but are the STAC assets using `az://` or `https://` prefixes ? I don't think rasterio can recognize https files being stored in azure which could explain why setting `AZURE_STORAGE_CONNECTION_STRING` would not have any influence.

If ☝️ is true then you have multiple choice:
- change the asset's url to `az://`  (I've done this in https://github.com/vincentsarago/MAXAR_opendata_to_pgstac/discussions/1)
- add a custom code in your raster service to translate asset's url to `az://` prefix or add the connection string.

**Reply by charalamm** [reply]
Hello @vincentsarago. Thanks for looking into it. In the STAC asset href I am using `az://`.

To add some more input, I see the same behavior in the local docker-compose setup when the only thing I do is to add the `AZURE_STORAGE_CONNECTION_STRING` environment variable.

**Reply by vincentsarago** [reply]
so just to be clear:
- your assets use `az://` prefix
- you have `AZURE_STORAGE_CONNECTION_STRING` set in your env

when ☝️, eoapi (titiler-pgstac) doesn't work while a simple rasterio code works?

**Reply by charalamm** [reply]
Yes that is correct

**Reply by charalamm** [reply]
I am running the following in production and it even returns a correct response. 
```
from rio_tiler.io import Reader
r = Reader("az://data/raster_file.tif")
r.dataset.read()
```
I tried that because it seems like `Reader` is class that actually reads the raster file.
